File: qhse-frontend/src/app/shared/models/resultat-global-response.ts
import { ResultatKpiResponse } from './resultat-kpi-response';

export interface ResultatGlobalResponse {
  importSessionId: number;
  periodeN1: number;
  periodeN: number;
  resultats: ResultatKpiResponse[];
  nombreFaible: number;
  nombreModere: number;
  nombreCritique: number;
  analyseGlobaleIa?: string;
}

File: QHSEAnalytics/src/main/java/com/QHSEAnalytics/shared/dto/response/ResultatGlobalResponse.java
package com.QHSEAnalytics.shared.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ResultatGlobalResponse {
    private Long importSessionId;
    private Integer periodeN1;
    private Integer periodeN;
    private List<ResultatKpiResponse> resultats;
    private long nombreFaible;
    private long nombreModere;
    private long nombreCritique;
    private String analyseGlobaleIa;
}
